### ISSUE TITLE
Export/Import Profile data

### ISSUE THREAD

**Issue by totalfrenzy** [sarahavilov/autofillforms-e10s#7]
Hi,

I'm really interested in an import/export feature to share the tool and profiles with my team.

In the meantime, I've been hunting for where the data is actually saved, but I've come up blank. What file do I need to copy across systems to keep the profiles?

Cheers

**Comment by sarahavilov** [comment]
You need to copy the extension's preferences stored in about:config. 
Find the addon id and you can filter-out the profile entries.

I am going to add the import/export feature soon. Still tying to make 
sure this new extension is working fine on all possible forms. Have you 
had any issues this the add-on?

**Comment by totalfrenzy** [comment]
Ah awesome. Just tested it and I was successfully able to copy a profile.

I've got a few other pieces of feedback:
- I've had issues extracting fields from forms - Does it only take the field you have selected? Having some feedback about how many fields, and what their labels were would be awesome
- Adding/Editing values and rules doesn't give much feedback. Something kind of js notification like this would be sweet: http://msroot.github.io/Notify.js/
- I've created a local test.html with fields for testing my profiles, but it doesn't seem to recognise any fields unless it's being served via a webserver. Would be great to allow file:// forms to be used as well as including a tutorial form for testing/learning purposes.
- In the old extension, I liked the right click to change profile, left click to fill out form
- I'm loving the Update profile option - it's awesome

Unfortunately the kinds of forms I'm using this on are using very poor labels so I'll be spending a bit of time creating custom regex field rules (Also I just noticed it's been misspelled in the Rules section as 'Filed').

Thanks so much! This plugin is already saving me heaps of time.

**Comment by sarahavilov** [comment]
Import and export features are ready. Please give the latest built a try and let me know how it works;
https://github.com/sarahavilov/autofillforms-e10s/blob/master/builds/packed/firefox.xpi?raw=true

Download the XPI and go to about:debuggin. Check the developer mode and point the browse button to the downloaded XPI.

**Comment by sarahavilov** [comment]
> I've had issues extracting fields from forms - Does it only take the field you have selected? 

No it should collect all fields in the active tab/window. If it is not working properly provide a sample URL with this issue.

> Having some feedback about how many fields, and what their labels were would be awesome

You should get a desktop notification whenever a profile is updated.

**Comment by sarahavilov** [comment]
> I've created a local test.html with fields for testing my profiles, but it doesn't seem to recognise any fields unless it's being served via a webserver. Would be great to allow file:// forms to be used as well as including a tutorial form for testing/learning purposes.

Thanks for the report. Latest dev supports "file://" scheme as well. I also added a sample form in
https://github.com/sarahavilov/autofillforms-e10s/tree/master/tests
You can request pull merge if yours is more field tpyes.

**Comment by sarahavilov** [comment]
> In the old extension, I liked the right click to change profile, left click to fill out form

Chrome as well as Firefox does not support right click anymore. I'll try to prevent using right or middle click in this extension. Changing profiles from panel seems to be easy enough.

**Comment by sarahavilov** [comment]
> Also I just noticed it's been misspelled in the Rules section as 'Filed'

fixed. thanks

**Comment by sarahavilov** [comment]
> Unfortunately the kinds of forms I'm using this on are using very poor labels so I'll be spending a bit of time creating custom regex field rules

If any of these new rules are useful for other users you may request a pull merge.

**Comment by sarahavilov** [comment]
> Adding/Editing values and rules doesn't give much feedback. Something kind of js notification like this would be sweet: http://msroot.github.io/Notify.js/

The library uses jQuery which is not necessary for this extension and is discouraged by AMO. We need to find a simple JS only lib.

**Comment by totalfrenzy** [comment]
> Import and export features are ready. Please give the latest built a try and let me know how it works

I've tried all the exporting options, but they don't seem to do anything. The import dialog seems to work.

> No it should collect all fields in the active tab/window

I've tried it on multiple browsers and computers. Seem to be the same for every website. Here's a sample:
http://www.phpform.org/formbuilder/view.php?id=in2i25brvtikg3ml4vtha5o5k3
After pressing extract rules 3 times, I now have element_4, element_5 and element_2

Just noticed the notifications, they were hiding on my other monitor!

> Changing profiles from panel seems to be easy enough.

Yeah it's not too bad at all. Sometimes I'm unable to select from the dropdown using my mouse for some reason. Seems to happen across computers. Using the keyboard works. 

> If any of these new rules are useful for other users you may request a pull merge.

Unfortunately a lot of the field labels are just numbers. I'll see if I come across any reusable fields. 

Thanks again!
